ui/src/features/freight-timeline/freight-content-item.tsx

```diff
@@ -54,21 +58,49 @@ export const FreightContentItem = (props: {
       overlay={overlay}
       title={title}
     >
-      <FontAwesomeIcon
-        icon={icon}
-        style={{ fontSize: '14px' }}
-        className={classNames('px-1', {
-          'mb-2': !horizontal,
-          'mr-2': horizontal
-        })}
-      />
-      {href ? (
-        <a target='_blank' className={linkClass}>
-          {_children}
-        </a>
-      ) : (
-        _children
-      )}
+      <div className='flex gap-2 items-center'>
+        <FontAwesomeIcon icon={icon} style={{ fontSize: '14px' }} className={classNames('px-1')} />
+        {props.artifactSource && (
+          <a
+            href={props.artifactSource}
+            className={classNames('text-blue-500', {
```

Comment:
Please don't use colors from tailwind. If you want to have a blue color link, use Typography.Link component.